pkg/actions/nodegroup/nodegroup.go

```diff
@@ -29,9 +31,28 @@ func New(cfg *api.ClusterConfig, ctl *eks.ClusterProvider, clientSet *kubernetes
 		cfg:          cfg,
 		clientSet:    clientSet,
 		wait:         waiters.Wait,
+		init: &eks.NodeGroupService{
+			Provider: ctl.Provider,
+		},
+		kubeProvider: ctl,
 	}
 }
 
+// MockKubeProvider can be used for passing a mock of the kube provider.
+func (m *Manager) MockKubeProvider(k eks.KubeProvider) {
+	m.kubeProvider = k
+}
+
+// MockNodeGroupService can be used for passing a mock of the nodegroup initialiser.
+func (m *Manager) MockNodeGroupService(ngSvc eks.NodeGroupInitialiser) {
+	m.init = ngSvc
+}
+
+// MockStackManager can be used for passing a mock of the stack manager.
+func (m *Manager) MockStackManager(sm manager.StackManager) {
+	m.stackManager = sm
+}
+
```

Comment:
should these be in an `export_test.go` file, to "hide" them from consumers of Manager? Example:
https://github.com/weaveworks/eksctl/blob/main/pkg/drain/export_test.go

the `export_test.go` is part of the package, so you can call it from within your tests. But because its in an `_test.go` file it doesn't get compiled by go when somebody uses this package externally

Reply:
very good idea, I'll move it!

Reply:
Moved it :)